### PR TITLE
update iris, new version of Flux, use MLDatasets added

### PR DESCRIPTION
**Pull request by luboshanus**
Small updates to iris.jl, using new version of Flux and loads data from MLDatasets.

My previous changes can be rejected. (https://github.com/FluxML/model-zoo/pull/369#discussion_r1031965305)

### CODE REVIEW

File: other/iris/Project.toml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [deps]
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
 Flux = "587475ba-b771-5e3f-ad9e-33799f191a9c"
+MLDatasets = "eb30cadb-4394-5ae3-aed4-317e484a6458"
 Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
 [compat]
-Flux = "0.11.4"
```
Comment (CarloLucibello) [inline other/iris/Project.toml:7]: why remove the compat bound? you should update this and add bounds for MLDatasets and DataFrames

File: other/iris/Project.toml [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+DataFrames = "1.4.3"
+Flux = "0.13.8"
+MLDatasets = "0.7.6"
 julia = "1.5"
```
Comment (CarloLucibello) [inline other/iris/Project.toml:12]:
```suggestion
julia = "1.6"
```

File: other/iris/iris.jl [hunk 3]
```diff
@@ -37,8 +43,10 @@ end
 
 
 function get_processed_data(args)
-    labels = Flux.Data.Iris.labels()
-    features = Flux.Data.Iris.features()
+
+    iris = Iris()
+    labels = iris.targets |> Matrix |> vec
+    features = iris.features |> Matrix |> permutedims
```
Comment (CarloLucibello) [inline other/iris/iris.jl:49]:
```suggestion
    iris = Iris(as_df=false)
    labels = iris.targets |> vec
    features = iris.features 
```

**Comment by luboshanus** [comment]
Thanks. Updated it!

**Comment by CarloLucibello** [comment]
looks fine, thanks!

**Pull request merged**